convert2rhel/subscription.py

```diff
@@ -135,65 +135,187 @@ def register_system():
     loggerinst.critical("Unable to register the system through subscription-manager.")
 
 
-def get_registration_cmd():
-    """Build a command for subscription-manager for registering the system."""
-    loggerinst.info("Building subscription-manager command ... ")
-    registration_cmd = ["subscription-manager", "register", "--force"]
-
-    loggerinst.info("Checking for activation key ...")
-    if tool_opts.activation_key:
-        # Activation key has been passed
-        # -> username/password not required
-        # -> organization required
-        loggerinst.info("    ... activation key detected: %s" % tool_opts.activation_key)
-
-        # TODO: Parse the output of 'subscription-manager orgs' and let the
-        # user choose from the available organizations. If there's just one,
-        # pick it automatically.
-        # Organization is required when activation key is used
+class RegistrationCommand(object):
+    def __init__(self, activation_key=None, org=None, username=None, password=None, server_url=None):
+        """
+        A callable that can register a system with subscription-manager.
+
+        :kwarg server_url: Optional url to the subscription-manager backend.
+            Useful when the customer has an on-prem subscription-manager instance.
+        :kwarg activation_key: subscription-manager activation_key that can be
+            used to register the system.  Org must be specified if this was given.
+        :kwarg org: The organization that the activation_key is associated with.
+            Required if activation_key is specified.
+        :kwarg username: Username to authenticate with subscription-manager.
+            Required if password is specified.
+        :kwarg password: Password to authenticate with subscription-manager.
+            Required if username is specified.
+
+        .. note:: Either activation_key and org or username and password must
+            be specfied.
+        """
+        self.cmd = "subscription-manager"
+        self.server_url = server_url
+
+        if activation_key and not org:
+            raise ValueError("org must be specified if activation_key is used")
+        self.activation_key = activation_key
+        self.org = org
+
+        self.password = password
+        self.username = username
+        if (password and not username) or (username and not password):
+            raise ValueError("username and password must be used together")
```

Comment:
I think I was trying to use empty lines to to group attributes being set with the code that was validating or otherwise manipulating them but I see where you're coming from.  This block is a bit longer than the other ones because of all the special interactions between password, username, and activation_key....

/me applies the change.